sync/evmdriver.go

```diff
@@ -93,9 +93,16 @@ func NewEVMDriver(
 	compatibilityChecker compatibility.CompatibilityChecker,
 ) (*EVMDriver, error) {
 	logger := log.WithFields("syncer", reorgDetectorID)
-	reorgSub, err := reorgDetector.Subscribe(reorgDetectorID)
-	if err != nil {
-		return nil, err
+
+	var reorgSub *reorgdetector.Subscription
+	var err error
+
+	// Only subscribe to reorg detector if it's not nil
+	if reorgDetector != nil {
```

Comment:
How about having some no-op reorg detector, so we don't have these checks if reorg detector is nil throughout the code?

Reply:
so basically who instead of passing `nil` we pass a new struct `NoOpReorgDetector`?
something like that?